libraries/hts221_driver/hts221.c

```diff
@@ -0,0 +1,301 @@
+
+
+#include "hts221.h"
+#include "string.h"
+#include <stdio.h>
+#include <stdint.h>
+
+#define DEBUG_LOG_MODULE_NAME "HTC221_DRIVER"
+#define DEBUG_LOG_MAX_LEVEL LVL_DEBUG
+#include "debug_log.h"
+
+/** Maximum I2C write transfer */
+#define MAX_WRITE_SIZE 16
+
+/** Maximum I2C read transfer */
+#define MAX_READ_SIZE 16
+
+#define HTS221_ADDR 0x5F
+
+#define AV_CONF_RESERVED_BITS (0xC0)   ///< Reserved bits in the AV_CONF register.
+#define CTRL_REG1_RESERVED_BITS (0x78) ///< Reserved bits in the CTRL_REG1 register.
+#define CTRL_REG2_RESERVED_BITS (0x7C) ///< Reserved bits in the CTRL_REG2 register.
+#define CTRL_REG3_RESERVED_BITS (0x3B) ///< Reserved bits in the CTRL_REG3 register.
+
+static i2c_conf_t m_i2c_conf = {
+    .clock = 100000,
+    .pullup = false,
+};
+
+/**@brief Configuration struct.
+ */
+typedef struct
+{
+    hts221_evt_handler_t evt_handler; ///< Event handler, called when data is ready.
+    bool enabled;                     ///< Driver enabled flag.
+    hts221_calib_t calib;             ///< Calibration struct.
+} htc221_t;
+
+/**@brief Stored configuration.
+ */
+static htc221_t htc221_sensor;
+
+/**
+    @brief      Write with I2C (function required by STMicroelectronics lib).
+    @param[in]  reg First register to write to.
+    @param[out] bufp Pointer in RAM to the data to be written.
+    @param[in]  len Number of registers (of 1 byte) to write.
+*/
+int32_t htc221_writeI2C(
+    uint8_t reg,
+    uint8_t *bufp,
+    uint16_t len)
+{
+
+    i2c_res_e res;
+    uint8_t tx[MAX_WRITE_SIZE + 1];
+
+    tx[0] = reg;
+    memcpy(&tx[1], bufp, len);
+    res = I2C_init(&m_i2c_conf);
```

Comment:
this value is always overwritten by line 70. Is there a missing if in line 62?

Reply:
The values `res` and `err`  are there to help us by error handling and to hold the error value that can be printed by DBUG-LOGs, But because i did this PR quickly i have not used them for error handling and DEBUG LOGs and therefore they are overwritten most of the time.
In the next round when i have time i will do that. But for the driver is working.